app/authenticators/torii.js

```diff
@@ -1,17 +1,55 @@
 import Torii from 'ember-simple-auth/authenticators/torii';
 import service from 'ember-service/inject';
+import fetch from 'fetch';
+import RSVP from 'rsvp';
+import config from 'wnyc-web-client/config/environment';
 
 export default Torii.extend({
   torii: service(),
+  store: service(),
+  session: service(),
 
   authenticate() {
-    return this._super(...arguments).then(data => {
-      return {
-        access_token: data.accessToken,
-        provider: data.provider,
-        expires_in: data.expiresIn,
-        user_id: data.userId
-      };
+    // Providers should return the normal data object, as well as
+    // an optional userAttrs object that can be provided to
+    // createRecord to create a new user
+    return new RSVP.Promise((resolve, reject) => {
+      this._super(...arguments).then(({data, userAttrs}) => {
+        let authData = {
+          access_token: data.accessToken,
+          provider: data.provider,
+          expires_in: data.expiresIn,
+          user_id: data.userId
+        };
+        // try to login
+        fetch(`${config.wnycAuthAPI}/v1/session`, {
+          method: 'GET',
+          headers: {
+            'Authorization': `Bearer ${data.accessToken}`,
+            'X-Provider': data.provider
+          }
+        })
+        .then((response) => {
+          // if we can log in, resolve
+          if (response && response.ok) {
+            resolve(authData);
+          // if we get a 401 Unauthorized, create a user
+          } else if (response && response.status === 401 && userAttrs) {
```

Comment:
Just checking: this is the part that the auth service is going to handle now, right? Perhaps this will remove the need to set this cross-concern 'isNewSocialUser' flag?